include/openthread/cli.h

```diff
@@ -153,6 +153,30 @@ void otCliUartAppendResult(otError aError);
  */
 void otCliPlatLogv(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, va_list aArgs);
 
+/**
```

Comment:
Is it possible to define a macro like `OT_UART_PLATFORM_HEADER` which provides such lock declarations? Just like `LWIP_HOOK_FILENAME` of https://github.com/yarrick/lwip/blob/master/src/core/ipv4/dhcp.c#L86. I think we can restrict all changes in `src/cli/cli_uart.cpp` only. This also gives platform developers more freedom.

Reply:
Move changes in this file into `src/cli/cli_uart.cpp`?